components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -121,18 +125,23 @@ func (n *Nats) SyncSubscription(sub *eventingv1alpha1.Subscription, cleaner even
 			}
 		}
 
-		for i := 0; i < subsConfig.MaxInFlightMessages; i++ {
-			// queueGroupName must be unique for each subscription and subject
+		for i := 0; i < subscriptionConfig.MaxInFlightMessages; i++ {
+			// queueGroupName must be unique for each sub and subject
 			queueGroupName := createKeyPrefix(sub) + string(types.Separator) + subject
 			natsSub, subscribeErr := n.connection.QueueSubscribe(subject, queueGroupName, callback)
 			if subscribeErr != nil {
-				log.Errorw("create NATS subscription failed", "error", err)
+				log.Errorw("creating NATS subscription failed", "error", subscribeErr)
 				return false, subscribeErr
 			}
 			n.subscriptions[createKey(sub, subject, i)] = natsSub
 		}
 	}
-	sub.Status.Config = subsConfig
+
+	if len(cleanSubjects) > 0 {
```

Comment:
Why do we need this check? why not always overwrite the `sub.Status.CleanEventTypes` with the cleaned ones?